### ISSUE TITLE
Minor defect in report subcommand when using a local path starting with s3

### ISSUE THREAD

**Issue by scotwk** [capitalone/cloud-custodian#561]
This is pretty minor issue and I wasn't sure if it warranted an issue report or not.  

I noticed that there was a discrepancy between how most policies determine whether to use s3 for the storage and how the `report` subcommand determines it.  If you pass a local file directory starting with `s3` to the `run` subcommand it works fine (and creates a local directory with the given name), but if you pass it to `report` it will give this error:

```
$ custodian report -c report.yml -s s3-foo-bar
Traceback (most recent call last):
  File "/Users/kkronenfeld/sixfeetup/cloud-custodian/bin/custodian", line 11, in <module>
    load_entry_point('c7n', 'console_scripts', 'custodian')()
  File "/Users/kkronenfeld/sixfeetup/cloud-custodian/c7n/cli.py", line 144, in main
    options.command(options)
  File "/Users/kkronenfeld/sixfeetup/cloud-custodian/c7n/commands.py", line 40, in _load_policies
    return f(options, policies)
  File "/Users/kkronenfeld/sixfeetup/cloud-custodian/c7n/commands.py", line 103, in report
    raw_output_fh=options.raw)
  File "/Users/kkronenfeld/sixfeetup/cloud-custodian/c7n/reports/csvout.py", line 83, in report
    policy.ctx.output.bucket,
AttributeError: 'DirectoryOutput' object has no attribute 'bucket'
```

I'll provide a fix on #536

**Comment by scotwk** [comment]
I forgot to link the commit, but this is fixed in PR #536 

This works now:

```
$ custodian run -c report.yml -s s3-foo-bar
2016-11-01 16:58:01,607: custodian.policy:INFO Running policy ec2-running-instances resource: ec2 region:us-east-1 c7n:0.8.19.7
2016-11-01 16:58:01,607: custodian.resources.ec2:INFO Filtered from 1 to 1 ec2
2016-11-01 16:58:01,607: custodian.policy:INFO policy: ec2-running-instances resource:ec2 has count:1 time:0.00

$ custodian report -c report.yml -s s3-foo-bar
action-date,instance-id,name,instance-type,launch,vpc-id,ip-addr,asv,env,owner
2016-11-01,i-09c1c849ee8955035,c7n-test-ec2,t2.micro,2016-10-25T13:08:35+00:00,vpc-d2d616b5,172.31.27.131,,,
```

**Comment by kapilt** [comment]
good catch, thanks
